backend/kernelCI_app/views/hardwareDetailsView.py

```diff
@@ -373,9 +374,11 @@ def sanitize_records(self, records, trees: List, is_all_selected: bool):
                 processed_builds.add(build_id)
                 continue
 
-            should_process_test = self.test_in_filter(test_filter_key, record)
+            should_process_test = not record["id"] in processed_tests \
+                and self.test_in_filter(test_filter_key, record)
```

Comment:
Equivalent code with better readability (and follows black formatter):
```suggestion
            should_process_test = (
                self.test_in_filter(test_filter_key, record)
                and record["id"] not in processed_tests
            )
```

Reply:
It's good to remember that using `not it` together as suggested by Marcelo (in the cases where it makes sense) is generally better for readability